atum/src/main/scala/za/co/absa/atum/core/MeasurementProcessor.scala

```diff
@@ -72,80 +158,18 @@ class MeasurementProcessor(private var measurements: Seq[Measurement]) {
 
   /** The method maps string representation of control type to measurement function.  */
   private def getMeasurementFunction(measurement: Measurement): MeasurementFunction = {
```

Comment:
This method was split into two (private wrapper with parsing string-based control values and the shared-within-atum part).

The rest of the `MeasurementProcessor` methods are the same, just moved between the class and the object.